armi/reactor/reactors.py

```diff
@@ -2262,42 +2251,54 @@ def processLoading(self, cs):
                 "Please make sure that this is intended and not a input error."
             )
 
-        # if we have these params loaded from the database do not overwrite,
-        # otherwise initialize them
-        if not self.p.referenceBlockAxialMesh:
+        # TODO: There is an inconsistency with makeAxialSnapList when detailedAxialExpansion: False
+        # In a standard run from the blueprints:
+        #     - All assemblies (blueprints included) have their b.p.topIndex
+        #       set in reference to self.refAssem.getAxialMesh().
+        # In a database load run:
+        #     - All assemblies in the core have their b.p.topIndex as set in the standard run.
+        #     - The blueprints assemblies have the b.p.topIndex set in reference to the assembly with the finest mesh.
+        #       This guarantees functionality of makeAxialSnapList, but, if the assembly with the finest mesh is
+        #       not a fuel assembly, assembly.setBlockMesh may produce unexpected results.
+        if dbLoad:
+            # reactor.blueprints.assemblies need to be populated
+            # this normally happens during armi/reactor/blueprints/__init__.py::_prepConstruction
+            # but for DB load, this is not called so it must be here.
+            # pylint: disable=protected-access
+            self.parent.blueprints._prepConstruction(cs)
+            if not cs["detailedAxialExpansion"]:
+                # Apply mesh snapping for self.parent.blueprints.assemblies
+                # To guarantee mesh snapping will function, makeAxialSnapList should be in reference
+                # to the assembly with the finest mesh as defined in the blueprints.
```

Comment:
you could further expand in comments that core assemblies have this param stored from the database load.

Reply:
Addressed https://github.com/terrapower/armi/pull/797/commits/7df5b13b56fe5ec2afb5ffa07487001a816833d2